src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2527,16 +2527,16 @@ private static unsafe int ConvertToBase64Array(char* outChars, byte* inData, int
 
         private static int ToBase64_CalculateAndValidateOutputLength(int inputLength, bool insertLineBreaks)
         {
-            long outlen = ((long)inputLength) / 3 * 4;          // the base length - we want integer division here.
-            outlen += ((inputLength % 3) != 0) ? 4 : 0;         // at most 4 more chars for the remainder
+            // the base length - we want integer division here, at most 4 more chars for the remainder
+            long outlen = ((long)inputLength + 2) / 3 * 4;
 
             if (outlen == 0)
```

Comment:
Worth adding an assert that outlen is either 0 or 4?

Reply:
`outlen` will be the actual base64 encoded length, so >= 4 iif inputLength > 0.
`Debug.Assert(outlen % 4 == 0)` would be possible, but I don't see any value in adding that assert.

Reply:
Ah, I see